rrule/Cargo.toml

```diff
@@ -43,4 +47,11 @@ cli-tool = ["clap"]
 serde = ["serde_with", "chrono/serde", "chrono-tz/serde"]
 
 # Allows EXRULE's to be used in the `RRuleSet`.
-exrule = []
\ No newline at end of file
+exrule = []
+
+# Allows to use WASM
+wasm = []
+
+[lib]
+#[cfg(feature = "wasm")]
+crate-type = ["cdylib", "rlib"] 
```

Comment:
I'm not sure. In my tests, it wasn't necessary. What do you think?

Reply:
Should contain the default "lib" as well. I will need to read up on this setting more before merging this PR

Reply:
Does it build with just "lib"? Why was "lib" replaced with ["cdylib", "rlib"]. I am not that familiar with how WASM internals work and don't want to merge anything I don't fully understand.

Reply:
@fmeringdal @antoinepairet 

Assuming in our context the below command as **Rust Build**:

```shell
cargo build
```

And the below command as **Wasm Build**:

```shell
wasm-pack build --release --target web --features "wasm"
```

We can analyze the different outputs:

1.  Using the below config (Cargo.toml file):

```
[lib]
crate-type = ["lib"] 
```

 1.a) the **Rust Build** works.

 1.b) the **Wasm Build** fails:

```shell
Error: crate-type must be cdylib to compile to wasm32-unknown-unknown. Add the following to your Cargo.toml file:

[lib]
crate-type = ["cdylib", "rlib"]
Caused by: crate-type must be cdylib to compile to wasm32-unknown-unknown. Add the following to your Cargo.toml file:

[lib]
crate-type = ["cdylib", "rlib"]
```


2) Using the below config (Cargo.toml file):

```
[lib]
crate-type = ["lib", "cdylib", "rlib"] 
```

2.a) the **Rust Build** and **Wasm Build** command fail and return the same output:

```shell
    Blocking waiting for file lock on build directory
warning: output filename collision.
The lib target `rrule` in package `rrule v0.10.0 (/Users/douglasmachado/Documents/Rust/Projects/rust-rrule/rrule)` has the same output filename as the lib target `rrule` in package `rrule v0.10.0 (/Users/douglasmachado/Documents/Rust/Projects/rust-rrule/rrule)`.
Colliding filename is: /Users/douglasmachado/Documents/Rust/Projects/rust-rrule/target/debug/deps/librrule.rlib
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The lib target `rrule` in package `rrule v0.10.0 (/Users/douglasmachado/Documents/Rust/Projects/rust-rrule/rrule)` has the same output filename as the lib target `rrule` in package `rrule v0.10.0 (/Users/douglasmachado/Documents/Rust/Projects/rust-rrule/rrule)`.
Colliding filename is: /Users/douglasmachado/Documents/Rust/Projects/rust-rrule/target/debug/librrule.rlib
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
thread 'main' panicked at 'assertion failed: mtimes.insert(output.clone(), mtime).is_none()', src/cargo/core/compiler/fingerprint/mod.rs:1096:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

3) Using the below config (Cargo.toml file):

```
[lib]
crate-type = ["cdylib", "rlib"] 
```

3.a) the **Rust Build** and **Wasm Build** command build successfully.

Reply:
Thanks for the context @dgmachado . I think the wasm build is fine, my main concern is the implications of the Rust build and how it might affect applications relying on this library. Before resolving this discussion I will have to do a deeper dive into WASM myself to understand what is going on and why this is required. Sorry for holding up this PR, but I can't merge anything that I don't fully understand as I have to maintain this.